File: 백준/Silver/2003. 수들의 합 2/수들의 합 2.java
import java.io.BufferedReader;
import java.io.IOException;
import java.io.InputStreamReader;
import java.util.StringTokenizer;

public class Main {
    public static void main(String[] args) throws IOException {
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
        StringTokenizer st = new StringTokenizer(br.readLine(), " ");

        int n = Integer.parseInt(st.nextToken());
        int m = Integer.parseInt(st.nextToken());
        int cnt = 0;

        st = new StringTokenizer(br.readLine(), " ");
        int[] arr = new int[n + 1];
        for (int i = 1; i <= n; i++) {
            arr[i] = Integer.parseInt(st.nextToken());
        }
        int[] sumArr = new int[n + 1];
        sumArr[1] = arr[1];
        for (int i = 2; i <= n; i++) {
            sumArr[i] = sumArr[i - 1] + arr[i];
        }

        for (int i = 0; i < n; i++) {
            for (int j = i + 1; j < n + 1; j++) {
                if (sumArr[j] - sumArr[i] == m)
                    cnt++;
            }
        }

        System.out.println(cnt);
    }
}

File: 백준/Silver/2003. 수들의 합 2/수들의 합 2.py
import sys

line = sys.stdin.readline().strip().split(' ')
n = int(line[0])
m = int(line[1])

arr = list()
line = sys.stdin.readline().strip().split(' ')
for i in range(n):
    arr.append(int(line[i]))

cnt, i, j = 0, 0, 0
cur_sum = arr[0]
while j < n:
    if cur_sum == m:
        cnt += 1
        j += 1
        if j == n:
            break
        cur_sum += arr[j]
    elif cur_sum < m:
        j += 1
        if j == n:
            break
        cur_sum += arr[j]
    else:
        cur_sum -= arr[i]
        i += 1

print(cnt)